### ISSUE TITLE
Add tests for FactorSet using different Factors

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#740]
Currently the tests are written only considering `DiscreteFactor`.

**Comment by raghavg7796** [comment]
@ankurankan Can I work on this?

**Comment by ankurankan** [comment]
@raghavg7796 Sure go ahead. But this might have some effects on other parts of the code and some tests might start failing.

**Comment by raghavg7796** [comment]
@ankurankan Shouldn't first #733 be solved first?

**Comment by ankurankan** [comment]
@raghavg7796 Yup shouldn't be too difficult. You will just need to change a few if conditions.

**Comment by anuragShingare30** [comment]
@ankurankan Can I work on this issue??
